### PR TITLE
Split vex subcommand, support multiple configs

### PR DESCRIPTION
**Pull request by puerco**
This PR splits the `wolfictl vex` subcommand and moves the current implementation to `wolfictl vex packages` in order to anticipate the upcoming `wolfictl vex sbom` subcommand. 

**Note:** This needs #8 and a rebase before merging.

It also adds support for generating VEX documents with data from multiple configuration files. So you can now generate a combined VEX doc by passing multiple melange configuration files:

```
wolfictl vex packages pkg/vex/testdata/curl.yml pkg/vex/testdata/git.yaml 
{
  "id": "vex-8a489e826e47a7a9bc8c4a7b37a2d8033360f06c29f65433d17794a881e2d735",
  "format": "text/vex",
  "author": "",
  "role": "",
  "timestamp": "2022-12-28T14:25:51.550582322-06:00",
  "statements": [
    {
      "vulnerability": "CVE-1234-5678",
      "products": [
        "pkg:apk/wolfi/git@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-daemon@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-email@2.39.0-r0?distro=wolfi"
      ],
      "status": "not_affected"
    },
    {
      "vulnerability": "CVE-2022-1111",
      "timestamp": "2022-12-23T01:28:16-05:00",
      "products": [
        "pkg:apk/wolfi/git@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-daemon@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-email@2.39.0-r0?distro=wolfi"
      ],
      "status": "under_investigation"
    },
    {
      "vulnerability": "CVE-2022-1111",
      "timestamp": "2022-12-23T02:11:57-05:00",
      "products": [
        "pkg:apk/wolfi/git@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-daemon@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-email@2.39.0-r0?distro=wolfi"
      ],
      "status": "not_affected",
      "justification": "component_not_present"
    },
    {
      "vulnerability": "CVE-2022-2222",
      "timestamp": "2022-12-24T01:28:16-05:00",
      "products": [
        "pkg:apk/wolfi/git@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-daemon@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-email@2.39.0-r0?distro=wolfi"
      ],
      "status": "under_investigation"
    },
    {
      "vulnerability": "CVE-2022-2222",
      "timestamp": "2022-12-24T02:12:49-05:00",
      "products": [
        "pkg:apk/wolfi/git@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-daemon@2.39.0-r0?distro=wolfi",
        "pkg:apk/wolfi/git-email@2.39.0-r0?distro=wolfi"
      ],
      "status": "affected",
      "action_statement": "action statement"
    },

```

### CODE REVIEW

File: pkg/cli/vex.go [hunk 1]
```diff
@@ -1,25 +1,67 @@
...
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	"github.com/wolfi-dev/wolfictl/pkg/vex"
 )
 
 func VEX() *cobra.Command {
-	var author, role string
+	cmd := &cobra.Command{
+		Use:   "vex",
+		Short: "Tools to generate VEX statments for Wolfi packages and images",
```
Comment (luhring) [inline pkg/cli/vex.go:16]:
```suggestion
		Short: "Tools to generate VEX statements for Wolfi packages and images",
```

File: pkg/cli/vex.go [hunk 2]
```diff
@@ -1,25 +1,67 @@
...
+		Long: `wolfictl vex: Tools to generate VEX statments for Wolfi packages and images
```
Comment (luhring) [inline pkg/cli/vex.go:17]:
```suggestion
		Long: `wolfictl vex: Tools to generate VEX statements for Wolfi packages and images
```

File: pkg/cli/vex.go [hunk 3]
```diff
@@ -28,22 +70,19 @@ func VEX() *cobra.Command {
...
 			if err != nil {
-				return err
+				return fmt.Errorf("creating VEX document: %w", err)
 			}
 
-			err = doc.ToJSON(os.Stdout)
-			if err != nil {
-				return err
+			if err := doc.ToJSON(os.Stdout); err != nil {
+				return fmt.Errorf("marshaling VEX document")
```
Comment (luhring) [inline pkg/cli/vex.go:79]: Should we add `...: %w` to show what the error was?

File: pkg/vex/vex.go [hunk 4]
```diff
@@ -115,6 +131,28 @@ func determineStatus(packageVersion string) vex.Status {
...
+	}
+
+	sort.Strings(hashes)
+	h := sha256.New()
+	if _, err := h.Write([]byte(strings.Join(hashes, ":"))); err != nil {
+		return "", fmt.Errorf("hashing config files")
+	}
+
+	// One hash to rule them all
+	return fmt.Sprintf("vex-%s", fmt.Sprintf("%x", h.Sum(nil))), nil
```
Comment (luhring) [inline pkg/vex/vex.go:157]: I noticed the errors returned in this function don't include the `err` value (e.g. via `%w`), is that intentional?

File: pkg/cli/vex.go (hunk 3 above)
Comment (puerco) [reply pkg/cli/vex.go:79]: Done!

File: pkg/vex/vex.go (hunk 4 above)
Comment (puerco) [reply pkg/vex/vex.go:157]: Funny, I don't know why I forgot all the wrappings there. I've added them

**Comment by puerco** [comment]
Thanks for the review @luhring , I've pushed another commit with changes addressing you comments

**Review by luhring — APPROVED** [review]
LGTM!

Thought for later, maybe `package` instead of `packages` for the subcommand's name? Less to type, consistent with the singular `sbom`, and the command can be used for a single package.. WDYT?

**Comment by puerco** [comment]
> maybe package instead of packages 

Sounds good to me, I will rename it in a follow up :)

**Pull request merged**